src/main/java/de/zalando/zmon/scheduler/ng/scheduler/Scheduler.java

```diff
@@ -51,13 +56,15 @@ public Scheduler(AlertRepository alertRepo, CheckRepository checkRepo, EntityRep
         this.entityRepo = entityRepository;
         this.queueSelector = queueSelector;
         this.schedulerConfig = schedulerConfig;
-        this.metrics = metrics;
         this.schedulerMetrics = new SchedulerMetrics(metrics);
 
         taskSerializer = new JavaCommandSerializer(schedulerConfig.getTaskSerializer());
 
-        service = new ScheduledThreadPoolExecutor(schedulerConfig.getThreadCount(), new CustomizableThreadFactory("scheduler-pool"));
-        shortIntervalService = new ScheduledThreadPoolExecutor(schedulerConfig.getThreadCount(), new CustomizableThreadFactory("scheduler-pool-fast"));
+        service = new ScheduledThreadPoolExecutor(schedulerConfig.getThreadCount(), new CustomizableThreadFactory("sc-pool-"));
+        service.setRemoveOnCancelPolicy(true);
```

Comment:
This dequeues before end of interval and not on next execution which is in
theory more what we want.

But it is hard to find from manual how to correctly unschedule as .cancel()
is confusing not 100% clear.

On Friday, 1 July 2016, Henning Jacobs <notifications@github.com> wrote:

> In src/main/java/de/zalando/zmon/scheduler/ng/scheduler/Scheduler.java
> <https://github.com/zalando-zmon/zmon-scheduler/pull/49#discussion_r69355455>
> :
>
> >          this.schedulerMetrics = new SchedulerMetrics(metrics);
> >
> >          taskSerializer = new JavaCommandSerializer(schedulerConfig.getTaskSerializer());
> >
> > -        service = new ScheduledThreadPoolExecutor(schedulerConfig.getThreadCount(), new CustomizableThreadFactory("scheduler-pool"));
> > -        shortIntervalService = new ScheduledThreadPoolExecutor(schedulerConfig.getThreadCount(), new CustomizableThreadFactory("scheduler-pool-fast"));
> > +        service = new ScheduledThreadPoolExecutor(schedulerConfig.getThreadCount(), new CustomizableThreadFactory("sc-pool-"));
> > +        service.setRemoveOnCancelPolicy(true);
>
> what is this change doing?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/zalando-zmon/zmon-scheduler/pull/49/files/6d177f277783377ed855be1278aba194665c9a25#r69355455>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABYGQ_MQ9xxB91LYgNVDSr__S03VO0Ydks5qRYCIgaJpZM4JDQ4c>
> .
>